### ISSUE TITLE
When a binding is mistyped, a unrelated error happens

### ISSUE THREAD

**Issue by luizperes** [luizperes/cantor#11]
```Haskell
let succ  => x in Universe: x + 1
let nfold => (n, f, x) in Universe:
          [ x                          n = 0     ]
          [ nfold . (n-1, f, f . x)    otherwise ]

# mistyping `acc` rather than `succ`
do nfold . (10, acc, 0)
```

*Expected behaviour*:
`acc is not a valid binding name`

*Actual behaviour*:
`f is not a valid binding name`

**Comment by luizperes** [comment]
Fixed on #12

**Comment by luizperes** [comment]
It should probably be still happening for nested bindings. Needs double-check.
